src/pre_processing/extract_landscape.py

```diff
@@ -0,0 +1,165 @@
+    logger.info(vec)
+    win.add_overlay(dets)
+    dlib.hit_enter_to_continue()
+
+
+def get_facial_vectors(specific_part=None, file_path=DATA_CSV_FILE, only_train_data=False,
+                       only_test_data=False, load_cached=False):
+    """
+    TODO: extract_only_specific_parts
+
+    A list of numpy arrays containing position of specific positions of a face. if no face found a [62,2]
+    zero vector is placed in matrix
+
+    Keyword Arguments:
+        specific_part -- can specify to return vectors for specified facial features:
+                         [RIGHT_BROW, LEFT_BROW, NOSE, RIGHT_EYE, LEFT_EYE, MOUTH]
+        only_test_data -- returns facial vectors for only training data
+        only_test_data -- returns facial vectors for only test data
+        load_cached -- returns image matrix from file if stored, saves processing images again
+
+    :return numpy array [n, 62, 2], where n is number of photos analysed
+    """
+
+    def _create_correct_size_facial_matrix(only_train_data, only_test_data):
+        if only_train_data:
+            facial_featutes_matrix = np.zeros([TRAIN_END_POINT, 68, 2])
+        elif only_test_data:
+            facial_featutes_matrix = np.zeros([PUBLIC_TEST_END_POINT - TRAIN_END_POINT, 68, 2])
+        else:
+            facial_featutes_matrix = np.zeros([PRIVATE_TEST_END_POINT, 68, 2])
+        return facial_featutes_matrix
+
+    def _save_processed_image_matrix(image_matrix, only_train_data, only_test_data):
+        if only_test_data:
+            file_name = TEST_DATA_CACHED_PATH
+        elif only_train_data:
+            file_name = TRAIN_DATA_CACHED_PATH
+        else:
+            file_name = ALL_DATA_CACHED_PATH
+        logger.debug("Saving image_matrix to file {}".format(file_name))
+        np.save(file_name, image_matrix)
+
+    def _load_cached_image_matrix(only_train_data, only_test_data):
+        if only_test_data:
+            file_name = TEST_DATA_CACHED_PATH
+        elif only_train_data:
+            file_name = TRAIN_DATA_CACHED_PATH
+        else:
+            file_name = ALL_DATA_CACHED_PATH
+        logger.debug("Attempting to load cached file: {}".format(file_name))
+        return np.load(file_name)
+
+    if load_cached:
+        try:
+            image_matrix = _load_cached_image_matrix(only_train_data, only_test_data)
+            return image_matrix
+        except FileNotFoundError:
+            logger.debug("No file found, Processing data")
+
+    logger.debug("Getting facial vectors")
+    photos_matrix = _extract_photos_from_file(file_path, False, only_train_data, only_test_data)
+    facial_featutes_matrix = _create_correct_size_facial_matrix(only_train_data, only_test_data)
+    detector = dlib.get_frontal_face_detector()
+    predictor = dlib.shape_predictor(PREDICTOR_PATH)
+    faces_found = 0
+    faces_not_found = 0
+    logger.debug("Starting facial detection and feature extraction")
+    for i in range(len(photos_matrix)):
```

Comment:
argh, i see it's used later, up to you if you wanna do it that way or enumerate. I don't think anybody swings either way.